chcemvediet/apps/inforequests/views/attachment.py

```diff
@@ -27,15 +27,20 @@ def attachment_upload(request):
     return attachments_views.upload(request, session, download_url_func)
 
 @require_http_methods([u'HEAD', u'GET'])
-@login_required(raise_exception=True)
 def attachment_download(request, attachment_pk):
-    permitted = {
-            Session: Q(session_key=request.session.session_key),
-            Message: Q(inforequest__applicant=request.user),
-            WizardDraft: Q(owner=request.user),
-            InforequestDraft: Q(applicant=request.user),
-            Action: Q(branch__inforequest__applicant=request.user),
-            }
+    if request.user.is_anonymous():
+        permitted = {
+                Action: Q(branch__inforequest__published=True),
+                }
+    else:
+        permitted = {
+                Session: Q(session_key=request.session.session_key),
+                Message: Q(inforequest__applicant=request.user),
+                WizardDraft: Q(owner=request.user),
+                InforequestDraft: Q(applicant=request.user),
+                Action: Q(branch__inforequest__applicant=request.user) |
+                        Q(branch__inforequest__published=True),
```

Comment:
Pozor, toto zvereni neanonymizovanu prilohu aj ak user zverejnuje iba anonymizovane ziadosti. Podmienka by mala byt nieco ako: (ziadost patri userovi) OR (ziadost je zverejnena AND user ziadosti neanonymizuje)